### PR TITLE
FIX: Pull to refresh All

### PR DESCRIPTION
**Pull request by marcosrdz**

### CODE REVIEW

**Comment by GladosBlueWallet** [comment]
♫ This was a triumph. I'm making a note here: HUGE SUCCESS ♫

 [android in browser] https://appetize.io/app/eggvb06vkva9uu2bkb7r66h8k8?device=pixel4

[download apk](https://lambda-download-android-build.herokuapp.com/download/8150)

File: screen/wallets/list.js [hunk 1]
```diff
@@ -183,7 +183,7 @@ const WalletsList = () => {
       console.log('onSnapToItem', wallets.length === index ? 'NewWallet/Importing card' : index);
       if (wallets[index] && (wallets[index].timeToRefreshBalance() || wallets[index].timeToRefreshTransaction())) {
         console.log(wallets[index].getLabel(), 'thinks its time to refresh either balance or transactions. refetching both');
-        refreshAllWalletTransactions(index, false).finally(() => setIsLoading(false));
+        refreshAllWalletTransactions(false).finally(() => setIsLoading(false));
```
Comment (Overtorment) [inline screen/wallets/list.js:186]: if you want to update all walllets on pull-to-refresh you need to mess with `onRefresh`, not with `onSnapToItem`

File: blue_modules/storage-context.js [hunk 2]
```diff
@@ -102,15 +102,18 @@ export const BlueStorageProvider = ({ children }) => {
...
-      const balanceStart = +new Date();
-      await fetchWalletBalances(lastSnappedTo);
-      const balanceEnd = +new Date();
-      console.log('fetch balance took', (balanceEnd - balanceStart) / 1000, 'sec');
-      const start = +new Date();
-      await fetchWalletTransactions(lastSnappedTo);
-      const end = +new Date();
-      console.log('fetch tx took', (end - start) / 1000, 'sec');
+      for (const [i, value] of wallets.entries()) {
+        await BlueElectrum.waitTillConnected();
```
Comment (Overtorment) [inline blue_modules/storage-context.js:106]: `waitTillConnected` shouldn't be in cycle

File: blue_modules/storage-context.js [hunk 3]
```diff
@@ -102,15 +102,18 @@ export const BlueStorageProvider = ({ children }) => {
...
+        const balanceStart = +new Date();
+        console.log(`Updating all wallets transactions. Currently updating ${value.getID()}`);
+        await fetchWalletBalances(i);
```
Comment (Overtorment) [inline blue_modules/storage-context.js:109]: if youd read some code youd saw that calling `refreshAllWalletTransactions(undefined)` of just  `refreshAllWalletTransactions()` refreshes  *all* wallets

File: blue_modules/storage-context.js [hunk 4]
```diff
@@ -102,15 +102,18 @@ export const BlueStorageProvider = ({ children }) => {
...
+        const balanceEnd = +new Date();
+        console.log('fetch balance took', (balanceEnd - balanceStart) / 1000, 'sec');
+        const start = +new Date();
+        await fetchWalletTransactions(i);
+        const end = +new Date();
+        console.log('fetch tx took', (end - start) / 1000, 'sec');
+      }
```
Comment (Overtorment) [inline blue_modules/storage-context.js:116]: anyway, this method is to refresh *all transactions*, not  *all wallets*

**Review by Overtorment — CHANGES_REQUESTED** [review]
no code in your bugs

**Pull request closed without merging**